File: test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/FunctionApps/dotnet/Confluent/SingleKafkaTriggerQueueOutput.cs
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Kafka;
using Microsoft.Azure.WebJobs.Extensions.Storage;
using Microsoft.Extensions.Logging;

namespace Confluent
{
    public class SingleKafkaTriggerQueueOutput
    {
        // KafkaTrigger sample 
        // Consume the message from "topic" on the LocalBroker.
        // Add `ConfluentBrokerList` and `ConfluentCloudPassword` to the local.settings.json
        // For EventHubs
        // "ConfluentBrokerList": "{EVENT_HUBS_NAMESPACE}.servicebus.windows.net:9093"
        // "ConfluentCloudPassword":"{EVENT_HUBS_CONNECTION_STRING}
        [FunctionName("SingleKafkaTriggerQueueOutput")]
        [return: Queue("e2e-dotnet-single-confluent")]
        public static string Run(
            [KafkaTrigger("ConfluentBrokerList",
                          "e2e-kafka-dotnet-single-confluent",
                          Username = "ConfluentCloudUsername",
                          Password = "ConfluentCloudPassword",
                          Protocol = BrokerProtocol.SaslSsl,
                          AuthenticationMode = BrokerAuthenticationMode.Plain,
                          ConsumerGroup = "$Default")] KafkaEventData<string> kevent, ILogger log)
        {
            log.LogInformation($"C# Kafka trigger function processed a message: {kevent.Value}");
            return kevent.Value;
        }
    }
}

File: test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/FunctionApps/java/Confluent/src/main/java/com/function/SingleKafkaTriggerQueueOutput.java
package com.function;

import com.microsoft.azure.functions.annotation.*;
import com.microsoft.azure.functions.*;
import org.json.*;

public class SingleKafkaTriggerQueueOutput {
    /**
     * This function consume KafkaEvents on the localhost. Change the topic, ConfluentBrokerList, and consumerGroup to fit your enviornment.
     * The function is trigged one for each KafkaEvent
     * @param kafkaEventData
     * @param context
     */
    @FunctionName("SingleKafkaTriggerQueueOutput")
    public void runOne(
            @KafkaTrigger(name = "kafkaTrigger", 
                          topic = "e2e-kafka-java-single-confluent", 
                          brokerList = "ConfluentBrokerList",
                          username = "ConfluentCloudUsername",
                          password = "ConfluentCloudPassword",
                          authenticationMode = BrokerAuthenticationMode.PLAIN,
                          protocol = BrokerProtocol.SASLSSL,
                          consumerGroup="$Default") String kafkaEventData,
            @QueueOutput(name = "message", 
                         queueName = "e2e-java-single-confluent", 
                         connection = "AzureWebJobsStorage") OutputBinding<String> message,
            final ExecutionContext context) {
        context.getLogger().info(kafkaEventData);
        JSONObject kafkaEventJsonObj = new JSONObject(kafkaEventData);
        context.getLogger().info(kafkaEventJsonObj.getString("Value"));
        message.setValue(kafkaEventJsonObj.getString("Value"));
    }
}
